### ISSUE TITLE
Git#log performs slowly when following renames

### ISSUE THREAD

**Issue by dometto** [repotag/rjgit#38]
This, for a blob with a small history (3 commits, including 1 rename),
```ruby
jruby-9.1.8.0 :006 > repo.git.log("renamed.md", "HEAD", :follow => true)
```
Performs very, very slowly. Intuitively, I'd say JGit must be able to do this faster, so perhaps we should look at our own own implementation of rename following.

I see [this](https://github.com/repotag/rjgit/blob/master/lib/git.rb#L95) line has been commented out -- could using the `pathFilter` speed things up possibly?

**Comment by bartkamphorst** [comment]
> perhaps we should look at our own own implementation of rename following.

Makes sense to start there. 😉

**Comment by dometto** [comment]
Resolved in #51
